### PR TITLE
RHCLOUD-36959 | feature: improve logging for the Superkey worker

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
Many of the current logs provided way too much information which made monitoring the service really hard. The goal of these changes is to provide that extra information only when an error occurs or when the "debug" log level is enabled.

For the rest of the informational logs, we are fine with just following the status of the worker.

## Jira ticket
[[RHCLOUD-36959]](https://issues.redhat.com/browse/RHCLOUD-36959)

### CODE REVIEW

**Review by dehort — COMMENTED** [review]
Hey Mikel, looking good! 

A suggestion...instead of repeatedly adding the fields to the logger, how about creating and instance of the logger at the top of the method and adding fields to that instance as they become available.

Something like this:
```
log := l.Log.WithFields(logrus.Fields{"org_id": orgIdHeader})
// doing lots of cool stuff here...
// and here...
log.Errorf(`Logging the org_id here, even though its not on this line...look ma, no hands :)`, eventType)
```

This keeps you from repeatedly having to jam the same fields into the logger over and over and it decreases the risk of messing up the name of the field, etc.

I also like to build an instance of the logger and pass it along to other methods as they "work on the request" :smile: .  This helps you "stitch" together the logs in kibana and it keeps from having to repeatedly set the  fields...which is kinda gross (but logging is just kinda gross).  I have seen some code add the logger instance to a `context` and pass the context object around but that is gross too.  It ends up being a global and your code has to "dig" the logger out of the context object to use it.  It can be problematic if the code always expects the logger to be in the context, but the code was called with a context that does not have the logger.  Then the code blows up.  I usually see this in the tests / when writing tests.  You have to reverse engineer what all needs to be in the context to call the method.  Its kinda difficult and hard to follow.

**Comment by MikelAlejoBR** [comment]
Yes, you are right. I did not know how to really solve this problem, because passing the logger in the context indeed does not seem right, but also having to pass the logger the functions feels like the less evil... Especially because it feels like the context should be used somehow in these situations. So, after reading the following two articles...

- [Pitfalls of context values and how to avoid or mitigate them in Go ](https://www.calhoun.io/pitfalls-of-context-values-and-how-to-avoid-or-mitigate-them/)
- [Add value to a context without losing type safety ](https://www.willem.dev/articles/how-to-add-values-to-context/)

... I think I found inspiration on how to actually do it:

**log_context.go**
```go
// tenantIdCtxKeyType defines the type for the tenant's key that will ensure type safety when storing or fetching the
// variable to/from the context.
type tenantIdCtxKeyType string

// tenantIdCtxKey defines the key to be used to store the tenant's identifier.
const tenantIdCtxKey tenantIdCtxKeyType = "tenant_id"

// LogWithContext returns a logger with all the fields defined in the context.
func LogWithContext(ctx context.Context) *logrus.Entry {
	logFields := logrus.Fields{}

	if value, ok := ctx.Value(tenantIdCtxKey).(tenantIdCtxKeyType); ok {
		logFields["tenant_id"] = value
	}

	return Log.WithFields(logFields)
}

// WithTenantId adds the given tenant ID to the given context.
func WithTenantId(ctx context.Context, tenantId string) context.Context {
	return context.WithValue(ctx, tenantIdCtxKey, tenantId)
}
```

**some_other_go_code.go**
```
// Define the logging context.
logContext := logger.WithTenantId(context.Background(), request.TenantID)
logContext = logger.WithSourceId(logContext, request.SourceID)
logContext = logger.WithApplicationId(logContext, request.ApplicationID)

// Do more stuff ...
logger.LogWithContext(logContext).Infof("Something has been done")

// Further down the code we pass the log context so that the logger can print the values that we set in the context too.
someFunction(logContext, myParam1, myParam2)
```

This way we can effectively use the context as intended without losing type safety and without passing the logger around.
